### PR TITLE
Enhance testing for `document` package.

### PR DESCRIPTION
**Pull request by j0n3lson**
#### What type of PR is this?
 /kind cleanup

#### What this PR does / why we need it:

In support of #1016 this PR enhances and increase code coverage within `document` package.

### Fixes

* When a category (kind) has multiple note entries, ensure each entry is printed on a separate line.

* Remove leading unwanted characters from note markdowns. The default go template
    prepends a dash ("-") before each note.

### Deprecation:

* Removes `TestNoteCollection` and `TestRenderMarkdownTeamplateGoldenFile`. Both of these methods have been replaced with `TestDocument_RenderMarkdownTemplate` which tests the same functionality.

* Removes `sortKinds()` and associated test case. This method was deprecated in
PR#1148 when sorting was moved to `CreateDocument()`

### Enhancements:

* Add a test case for `CreateDocument()`.

* When testing template rendering, use `CreateDocument()` the same way a package user would. This makes it easier to test from the user's point of view.

#### Which issue(s) this PR fixes:

NONE

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by justaugustus** [comment]
/lgtm cancel
(since @j0n3lson mentioned there was a bug)

**Comment by j0n3lson** [comment]
/hold cancel

Ready to go. The bug was for entries not printing on their new line. I introduced the back a few commits back while trying to fix the "no-new-line-at-end-of-document-problem" which annoyingly doesn't seem possible in golang templates. I settled for `strings.TrimSuffix()` on the rendered template.

File: pkg/notes/document/document.go [hunk 1]
```diff
@@ -279,7 +281,7 @@ func (d *Document) RenderMarkdownTemplate(bucket, fileDir, templateSpec string)
 	if err := tmpl.Execute(&s, d); err != nil {
 		return "", errors.Wrapf(err, "rendering with template")
 	}
-	return s.String(), nil
+	return strings.TrimSuffix(s.String(), "\n"), nil
```
Comment (saschagrunert) [inline pkg/notes/document/document.go]:
```suggestion
	return strings.TrimSpace(s.String()), nil
```

Comment (j0n3lson) [reply pkg/notes/document/document.go]: Done.

File: pkg/notes/document/template.go [hunk 2]
```diff
@@ -71,12 +71,11 @@ filename | sha512 hash
...
+{{- if .Notes -}}
 ## Changes by Kind
-{{ range .}}
+{{ range .Notes}}
 ### {{.Kind | prettyKind}}
-{{range $note := .NoteEntries }} - {{$note}}{{end}}
-{{- end}}
-{{- end}}
-{{- /* This removes any extra line at the end. */ -}}
+{{range $note := .NoteEntries }}{{println " -" $note}}{{end}}
```
Comment (saschagrunert) [inline pkg/notes/document/template.go]: This does not work right now, if a note is already prefixed with `- ` then it will result in `- - Note`.

**Comment by saschagrunert** [comment]
@j0n3lson might give it a rebase to pull in the latest test-fixes

File: pkg/notes/document/template.go (hunk 2 above)
Comment (saschagrunert) [reply pkg/notes/document/template.go]: @j0n3lson do you see a way to fix this, maybe with a function?

Comment (saschagrunert) [reply pkg/notes/document/template.go:78]: Hey 👋, any news on this one? Otherwise I can have a look into it if you want.

**Comment by j0n3lson** [comment]
Hey Sasha,

I fixed it in my local side. I've not had a chance to look again in a bit.
I think I'll have time on Friday to rebase and push. I was only blocked on
making the test pass.

**Comment by saschagrunert** [comment]
> Hey Sasha, I fixed it in my local side. I've not had a chance to look again in a bit. I think I'll have time on Friday to rebase and push. I was only blocked on making the test pass.

Awesome, thank you for the update and keeping us posted.

**Comment by j0n3lson** [comment]
Heya @saschagrunert just crawled out from my rock. Mind taking a look now?

**Comment by justaugustus** [comment]
/lgtm
/approve

Thanks!

**Pull request merged**